src/main/scala/org/broadinstitute/dsde/workbench/gpalloc/service/GPAllocService.scala

```diff
@@ -141,11 +141,44 @@ class GPAllocService(protected val dbRef: DbReference,
     Future.successful(())
   }
 
+  def nukeProject(project: String, deleteInGoogle: Boolean = true): Future[Unit] = {
+    nukeProjectInternal(project, deleteInGoogle) {  maybeCreateNewProjects()  }
+  }
+
+  private def nukeProjectInternal(project: String, deleteInGoogle:Boolean = true)(op: => Unit): Future[Unit] = {
+    logger.info(s"attempting nuke of project $project")
+    val nuke = for {
+      _ <- dbRef.inTransaction { da => da.billingProjectQuery.deleteProject(project) }
+      _ <- if(deleteInGoogle) googleBillingDAO.deleteProject(project) else Future.successful(())
+    } yield ()
+    nuke.onComplete {
+      case Failure(e) =>
+        logger.error(s"surprise error nuking project $project because $e")
+      case Success(_) =>
+        logger.info(s"successful nukeProject of $project")
+        op
+    }
+    nuke
+  }
+
+  def nukeAllProjects(deleteInGoogle: Boolean = true): Future[Unit] = {
+    logger.info("nuking all projects")
+    val nukeAll = for {
+      allProjects <- dbRef.inTransaction { da => da.billingProjectQuery.listEverything() }
+      _ <- Future.traverse(allProjects) { rec => nukeProjectInternal(rec.billingProjectName, deleteInGoogle)(()) }
+    } yield ()
+    nukeAll.onComplete {
+      case _ => maybeCreateNewProjects()
+    }
+    //run the above future in the background because it's gonna take a while
+    Future.successful(())
+  }
+
```

Comment:
why not just:

```
def nukeProject() = nukeProjectInternal onComplete maybeCreate

def nukeAll() = for { <- traverse(nukeInternal) } onComplete maybeCreate
```

it's a bit confusing that you're passing in an op to do something afterwards in one instance, but then not using it in the other instance